src/ctl/v1/shard.go

```diff
@@ -195,6 +195,14 @@ func shardBalanceAdviceHandler(log *xlog.Log, proxy *proxy.Proxy, w rest.Respons
 		w.WriteJson(nil)
 		return
 	}
+	// Filter the global table.
+	router := proxy.Router()
+	shardKey, err := router.ShardKey(database, table)
+	if err == nil && shardKey == "" {
+		log.Warning("api.v1.balance.advice.return.nil.since.find.the.table.is.global.table")
+		w.WriteJson(nil)
```

Comment:
Here we can't return, we should only skip the global table and find the best partition table.
If the global table is larger, then we can't get the best table, always returns nil